src/app/features/course-info/course-info.component.ts

```diff
@@ -1,20 +1,52 @@
-import { Component, Input } from "@angular/core";
+import { Component, EventEmitter, OnInit } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
 import { ButtonLabels } from "@app/shared/constants/button-labels";
-interface Course {
-  id: string;
-  title: string;
-  description: string;
-  creationDate: string;
-  duration: number;
-  authors: string[];
-}
+import { mockedCoursesList } from "@app/shared/mocks/mocks";
+import { Location } from "@angular/common";
+import { CoursesStoreService } from "@app/services/courses-store.service";
+import { Course } from "@app/shared/models/course.model";
+
 @Component({
   selector: "app-course-info",
   templateUrl: "./course-info.component.html",
   styleUrls: ["./course-info.component.scss"],
 })
-export class CourseInfoComponent {
-  @Input() course!: Course;
+export class CourseInfoComponent implements OnInit {
+  course!: Course | undefined;
   btnTextBack: string = ButtonLabels.back;
-  // Use the names for the input `course`.
+  courseId: string | null = null;
+  courseAuthors!: string[];
+  authorNames!: string[];
+
+  constructor(
+    private route: ActivatedRoute,
+    private location: Location,
+    private courseStoreService: CoursesStoreService
+  ) {}
+
+  ngOnInit(): void {
+    this.courseId = this.route.snapshot.paramMap.get("id");
+
+    if (this.courseId) {
+      this.courseStoreService.getCourse(this.courseId).subscribe({
+        next: (course) => {
+          this.course = course.result;
+          this.courseAuthors = course.result.authors;
+        },
+      });
+      //this.getAuthorById();
+    }
+  }
+
+  getAuthorById(): void {
+    this.courseAuthors.forEach((id) => {
+      this.courseStoreService.getAuthorById(id).subscribe((author) => {
```

Comment:
Hmmm... This created a new subscription for each author associated with the course. So, at least one subscription is created with no upper limit. This approach imposes a performance issue in the application.
Can you refactor this to make it performant again? You can get all the authors and select the actual course authors. Also, make sure to unsubscribe.